CesiumGeospatial/test/TestEarthGravitationalModel96Grid.cpp

```diff
@@ -0,0 +1,225 @@
+#include <CesiumGeospatial/EarthGravitationalModel96Grid.h>
+#include <CesiumUtility/Math.h>
+
+#include <catch2/catch.hpp>
+
+#include <filesystem>
+
+using namespace CesiumGeospatial;
+using namespace CesiumUtility;
+
+struct Egm96TestCase {
+public:
+  Cartographic cartographicPosition;
+  double expectedHeight;
+
+  Egm96TestCase(Cartographic position, double height)
+      : cartographicPosition(position), expectedHeight(height) {}
+};
+
+// Long, Lat values calculated randomly and paired with expected results from
+// https://www.unavco.org/software/geodetic-utilities/geoid-height-calculator/geoid-height-calculator.html
+const Egm96TestCase testCases[] = {
+    Egm96TestCase(
+        Cartographic::fromDegrees(135.89012584487307, 11.046411138991914, 0),
+        57.79),
+    Egm96TestCase(
+        Cartographic::fromDegrees(179.78766535213848, -66.77911223257036, 0),
+        -57.37),
```

Comment:
You're testing both positive and negative values, so my fears about sign extension are unfounded. Up to you if you like the code I suggested better anyway.